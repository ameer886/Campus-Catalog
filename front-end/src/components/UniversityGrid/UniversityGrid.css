.Centering {
  display: flex;
  align-items: center;
  justify-content: center;
}

.UniversityGridContainer {
  display: grid;
  width: 720px;

  grid-template-columns: repeat(3, 1fr);

  grid-gap: 12px;

  align-items: center;
  justify-content: center;
}

.UniversityCard {
  height: 240px;

  border: 1px solid black;
  border-radius: 5px;

  background-color: #ddd;
  color: #222;
}
