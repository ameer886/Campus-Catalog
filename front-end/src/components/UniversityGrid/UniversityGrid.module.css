/* Generic Containers */
.Centering {
  display: flex;
  align-items: center;
  justify-content: center;
}

.UniversityGridContainer {
  display: grid;
  width: 75%;

  grid-template-columns: repeat(3, 1fr);

  grid-gap: 12px;

  align-items: center;
  justify-content: center;
}

.UniversityCard {
  min-height: 240px;

  border: 1px solid black;
  border-radius: 5px;

  background-color: #ddd;
  color: #222;
}

.CardLink {
  padding: 0;
}

/* Specific Card elements */
.UniversityName {
  font-size: 20px;
  font-style: bold;
  color: darkblue;
  margin-bottom: 0;
}

.LeftRightPair {
  width: 100%;
  display: flex;
  flex-flow: row wrap;
}

.Left {
  text-align: left;
  width: 65%;
  font-weight: bold;
  margin: 2px 0px 2px 2px;
}

.InState {
  color: blue;
}

.OutState {
  color: orange;
}

.Tuition {
  text-align: right;
  width: 33%;
  margin: 2px 2px 2px 0px;
}

.RankingText {
  color: red;
  font-size: 18px;
  font-style: bold;
}

.Ranking {
  font-size: 18px;
  font-style: bold;
  color: black;
}
