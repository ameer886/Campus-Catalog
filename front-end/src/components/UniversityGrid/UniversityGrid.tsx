import React from 'react';
// import './UniversityGrid.css';
import styles from './UniversityGrid.module.css';

import type { UniversityRowType } from '../../views/Universities/UniversitiesPage';
import { Nav } from 'react-bootstrap';
import { useLocation } from 'react-router-dom';
import { formatNumberToMoney } from '../../utilities';

type CardProps = {
  university: UniversityRowType;
};

const UniversityCard: React.FunctionComponent<CardProps> = ({
  university,
}: CardProps) => {
  const location = useLocation();
  const curPath = location.pathname;

  return (
    <Nav.Link
      className={styles.CardLink}
      href={curPath + '/id=' + university.id}
    >
      <div className={styles.UniversityCard}>
        <h2 className={styles.UniversityName}>
          {university.univ_name}
        </h2>
        {university.state && <p>Located in {university.state}</p>}

        <div className={styles.LeftRightPair}>
          <p className={styles.InState + ' ' + styles.Left}>
            In-State Tuition:
          </p>
          <p className={styles.Tuition}>
            {formatNumberToMoney(university.tuition_in_st)}
          </p>
        </div>
        <div className={styles.LeftRightPair}>
          <p className={styles.OutState + ' ' + styles.Left}>
            Out-of-State Tuition:
          </p>
          <p className={styles.Tuition}>
            {formatNumberToMoney(university.tuition_out_st)}
          </p>
        </div>

        <div className={styles.LeftRightPair}>
          <p className={styles.Left + ' ' + styles.RankingText}>
            Ranking:
          </p>
          <p className={styles.Tuition + ' ' + styles.Ranking}>
            {university.rank}
          </p>
        </div>
      </div>
    </Nav.Link>
  );
};

type UniversityGridProps = {
  cards: Array<UniversityRowType>;
};

/*
 * Grid for displaying universities
 * Each row will have at most 3 equally spaced cards
 * Each card should have info for a single university
 */
const UniversityGrid: React.FunctionComponent<UniversityGridProps> =
  ({ cards }: UniversityGridProps) => {
    return (
      <div className={styles.Centering}>
        <div className={styles.UniversityGridContainer}>
          {cards.map((university, index) => (
            <UniversityCard university={university} key={index} />
          ))}
        </div>
      </div>
    );
  };

export default UniversityGrid;
