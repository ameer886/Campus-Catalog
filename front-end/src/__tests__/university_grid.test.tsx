import React from 'react';
import { render, screen } from '@testing-library/react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { MemoryRouter } from 'react-router-dom';

import type { UniversityRowType } from '../views/Universities/UniversitiesPage';
import UniversityGrid from '../components/UniversityGrid/UniversityGrid';

configure({ adapter: new Adapter() });

const EXAMPLE_ROWS: Array<UniversityRowType> = [
  {
    acceptance_rate: 0.7366,
    avg_cost_attendance: 24495.0,
    city: 'Birmingham',
    ownership_id: 'Public',
    rank: 148,
    state: 'AL',
    tuition_in_st: 8568,
    tuition_out_st: 20400,
    univ_id: '100663',
    univ_name: 'University of Alabama at Birmingham',

    id: '100663',
  },
  {
    acceptance_rate: 0.3182,
    avg_cost_attendance: 26254.0,
    city: 'Austin',
    ownership_id: 'Public',
    rank: 38,
    state: 'TX',
    tuition_in_st: 10824,
    tuition_out_st: 38326,
    univ_id: '228778',
    univ_name: 'The University of Texas at Austin',

    id: '228778',
  },
  {
    acceptance_rate: 0.0464,
    avg_cost_attendance: 73485.0,
    city: 'Cambridge',
    ownership_id: 'Private Non-Profit',
    rank: 2,
    state: 'MA',
    tuition_in_st: 51925,
    tuition_out_st: 51925,
    univ_id: '166027',
    univ_name: 'Harvard University',

    id: '166027',
  },
];

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: '/universities',
  }),
}));

describe('University Grid Test Suite', () => {
  it('snapshot test', () => {
    const snap = shallow(<UniversityGrid cards={EXAMPLE_ROWS} />);
    expect(snap).toMatchSnapshot();
  });

  /*
   * Note: we cannot test the sorting here because it's tied to
   * the UniversitiesPage model page file. We'd have to mock the
   * axios backend to test that file, so for now the sorting will
   * have to go untested.
   */

  /*
   * Similarly, pagination is tied to the parent file so we won't
   * be able to test changing pages here
   */

  it('links have correct href', () => {
    render(
      <MemoryRouter initialEntries={['/amenities']}>
        <UniversityGrid cards={EXAMPLE_ROWS} />
      </MemoryRouter>,
    );

    for (let i = 0; i < EXAMPLE_ROWS.length; i++) {
      const aptLink = screen.getByRole('link', {
        name: EXAMPLE_ROWS[i].univ_name + ' link',
      });
      expect(aptLink).not.toBeNull();
      expect(aptLink.getAttribute('href')).toBe(
        `/universities/${EXAMPLE_ROWS[i].id}`,
      );
    }
  });
});
