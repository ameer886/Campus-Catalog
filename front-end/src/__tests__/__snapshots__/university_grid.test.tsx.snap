// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`University Grid Test Suite snapshot test 1`] = `ShallowWrapper {}`;
